File: solutions/erlang/bob/1/src/bob.erl
-module(bob).

-export([response/1]).

response(_String) ->
    Phrase = string:trim(_String),
    Length = string:len(Phrase),
    Is_Question = string:str(Phrase, "?") =:= Length,
    Has_Lower = string:len(string:join(
                  string:tokens(Phrase, "abcdefghijklmnopqrstuvwxyz"),
                  "" )) =/= Length,
    Has_Upper = string:len(string:join(
                  string:tokens(Phrase, "ABCDEFGHIJKLMNOPQRSTUVWXYZ"),
                  "" )) =/= Length,
    if Phrase =:= "" ->
            "Fine. Be that way!";
       Is_Question, not(Has_Lower), Has_Upper ->
            "Calm down, I know what I'm doing!";
       Is_Question ->
            "Sure.";
       Has_Upper, not Has_Lower ->
            "Whoa, chill out!";
       true ->
            "Whatever."
    end
.

File: solutions/fortran/bob/1/bob.f90
module bob
  implicit none
contains

  function hey(statement)
    character(100) :: hey
    character(len=*), intent(in) :: statement
    character(len=100) :: phrase
    character(len=100) :: answer
    integer :: size
    integer :: i
    integer :: caps

    phrase = statement
    size   = len(trim(phrase))
    caps   = 0

    do i = 1, size
       if ( iachar(phrase(i:i)) >= 97 .and. iachar(phrase(i:i)) <= 122 ) then
          caps = 0
          exit
       end if
       if ( iachar(phrase(i:i)) >= 65 .and. iachar(phrase(i:i)) <= 90 ) then
          caps = caps + 1
       end if
    end do

    if ( phrase == '' ) then
       answer = 'Fine. Be that way!'
    else if ( phrase(size:size) == '?' ) then
       if ( caps > 0 ) then
          answer = "Calm down, I know what I'm doing!"
       else
          answer = 'Sure.'
       end if
    else if ( caps > 0 ) then
       answer = 'Whoa, chill out!'
    else
       answer = 'Whatever.'
    end if

    hey = answer
  end function hey

end module bob
